lib/simple_form_attachments.rb

```diff
@@ -24,6 +25,15 @@ def self.dom_class(*args)
       prefix.compact.join('__')
     end
   end
+
+  def self.jquery_ui_asset(asset)
+    jquery_ui_rails_version = Gem.loaded_specs["jquery-ui-rails"].version
+    breaking_version = Gem::Version.new('6.0')
+    case
+    when jquery_ui_rails_version >= breaking_version then "jquery-ui/widgets/#{asset}"
```

Comment:
I like this solution, perhaps more explicit could be:

```ruby
case jquery_ui_rails_version
when 5...6 then "jquery-ui/#{asset}"
when 6...7 then "jquery-ui/widgets/#{asset}"
else raise "This version of jQuery UI is not supported"
end
```

Note the `...` instead of `..`, which do not match the end of range – `5..6` matches 6, while `5...6` does not match 6.

Reply:
That would indeed be nicer! But it seems we can get the version as either a string or a `Gem::Version`. There is a built in [comparison method](https://github.com/rubygems/rubygems/blob/fca24a695d2ee3a0bdb9702b752b34d44def6a53/lib/rubygems/version.rb#L330) in `Gem::Version`, but I'm not sure if we will make the switch case nicer by using that?

After looking at it again I think it's better to use the jquery ui version as specified in `Jquery::Ui::Rails::JQUERY_UI_VERSION`, so our method would look like this:

```RUBY
def self.jquery_ui_asset(asset)
  jquery_ui_version = Gem::Version.new(Jquery::Ui::Rails::JQUERY_UI_VERSION)
  breaking_version = Gem::Version.new('1.12.0')
  case
  when jquery_ui_version >= breaking_version then "jquery-ui/widgets/#{asset}"
  when jquery_ui_version < breaking_version then "jquery-ui/#{asset}"
  end
end
```

Otherwise we could use the `<=>` from `Gem::Version` and do something like this:

```RUBY
def self.jquery_ui_asset(asset)
  jquery_ui_version = Gem::Version.new(Jquery::Ui::Rails::JQUERY_UI_VERSION)
  breaking_version = Gem::Version.new('1.12.0')
  case breaking_version <=> jquery_ui_version
  when -1..0 then "jquery-ui/widgets/#{asset}"
  when 1 then "jquery-ui/#{asset}"
  end
end
```

Reply:
Sounds good to use `Gem::Version`. How about this then?

```ruby
jquery_ui_version = Gem::Version.new(Jquery::Ui::Rails::JQUERY_UI_VERSION)

case
when Gem::Dependency.new('', '< 1.12').match?(jquery_ui_version) then "jquery-ui/#{asset}"
when Gem::Dependency.new('', '>= 1.12').match?(jquery_ui_version) then "jquery-ui/widgets/#{asset}"
else raise "This version of jQuery UI is not supported"
end
```